doc/rtd/reference/datasources/nocloud.rst

```diff
@@ -12,6 +12,12 @@ provide configurations.
 Configuration Methods:
 ======================
 
+.. warning::
```

Comment:
I don't love putting this here, but I do think we've had enough confusion about this that it should be called out on this page. I'm open to suggestions for a better location.

Reply:
I'm not even 100% clear on what this warning block means? I think this is a fine location for it though. And a warning is definitely worthwhile to provide, especially if its a known confusion point!

Reply:
@holmanb could you please point me the exact change in 24.1? Some internal user did get caught by surprise by this change you talk about.

Reply:
@holmanb I have a further question for this change, could you please help to clarify this?
Just as @ani-sinha mentioned, we have some internal user did get caught by surprise by this change you talked about. 

It reports "DataSourceNoCloud.py[DEBUG]: Seed from https://..../userdata/ not supported by DataSourceNocloud" in cloud-init.log with below config in cloud-init-23.4(But no such issue before 23.1):

# cat << EOM > /etc/cloud/cloud.cfg.d/10_datasource.cfg
datasource_list: [NoCloud]
datasource:
  NoCloud:
    seedfrom: https://satellite.example.com/userdata/
EOM


Could I know how to adapt it after the change?
Thanks!

Reply:
@zhaohuijuan @ani-sinha sometimes comments like get missed when commenting on old merged PRs - github notifications are the only way that these messages get notified and since there is no queue for reviewing old comments comments like this are more likely to get lost. IRC would be more likely to get a response.

To answer your question, I don't think that this log message indicates an actual issue:
```

Reply:
Thanks @holmanb for the reminder and quick response. Seems I can not login the IRC #cloud-init ... So asking here.

It indeed broke the cloud-init implementation, not sure if  this log related. 

Per the latest cloud-init doc[1]: User data placed under /etc/cloud/ will not be recognized as a source of configuration data by the NoCloud datasource.
[1] https://cloudinit.readthedocs.io/en/latest/reference/datasources/nocloud.html

And the doc provides 4 configuration methods(e.g. command line), should we config NoCloud and seedfrom with one of the 4 methods?

But there is another example for NoCloud config[2], is it conflict with the above doc[1]?
[2] https://cloudinit.readthedocs.io/en/latest/reference/examples.html#configure-data-sources

Reply:
> Thanks @holmanb for the reminder and quick response. Seems I can not login the IRC #cloud-init ... So asking here.
> 
> It indeed broke the cloud-init implementation, not sure if  this log related. 

I don't think that this log is related, but it sounds like there may be an issue that should be reported. There have been lots of different changes in this part of the code over the last two years. Without logs of the working and non-working instances this may be difficult to understand.

Can you please file a bug with the relevant logs?

> Per the latest cloud-init doc[1]: User data placed under /etc/cloud/ will not be recognized as a source of configuration data by the NoCloud datasource.
> [1] https://cloudinit.readthedocs.io/en/latest/reference/datasources/nocloud.html
>
> And the doc provides 4 configuration methods(e.g. command line), should we config NoCloud and seedfrom with one of the 4 methods?
> 
> But there is another example for NoCloud config[2], is it conflict with the above doc[1]?
> [2] https://cloudinit.readthedocs.io/en/latest/reference/examples.html#configure-data-sources

This sounds like possibly confusion between user-data and system configuration, the doc referenced is not describing user-data I think.

Reply:
Ahh, you're referring to this:
```
    # these are optional, but allow you to basically provide a datasource

    # right here

    user-data: |

      # This is the user-data verbatim

    meta-data:

      instance-id: i-87018aed

      local-hostname: myhost.internal
```

Yes, we should probably update the doc wording. You can use this subkey, which is technically defined within cloud.cfg files.

Reply:
Thanks @holmanb 
I think you already involved in the issue: https://github.com/canonical/cloud-init/issues/5288, thanks for your clarification